### PR TITLE
feat: add LawCommIdentity + IdempotentOp for BitVec.[and|or|xor]

### PR DESCRIPTION
**Pull request by tobiasgrosser**
As these instances seemingly require explicit arguments, this PR also makes some arguments explicit.

### CODE REVIEW

**Comment by tobiasgrosser** [comment]
I feel we really need to understand better under which conditions arguments should be explicit or implicit and then should standardize this across the BitVec library.

File: src/Init/Data/BitVec/Lemmas.lean [hunk 1]
```diff
@@ -710,14 +710,20 @@ theorem or_comm (x y : BitVec w) :
...
 instance : Std.Commutative (fun (x y : BitVec w) => x ||| y) := ⟨BitVec.or_comm⟩
 
-@[simp] theorem or_self {x : BitVec w} : x ||| x = x := by
+@[simp] theorem or_self (x : BitVec w) : x ||| x = x := by
   ext i
   simp
 
-@[simp] theorem or_zero {x : BitVec w} : x ||| 0#w = x := by
+instance : Std.IdempotentOp (α := BitVec n) (· ||| · ) where
+  idempotent := BitVec.or_self
```
Comment (semorrison) [inline src/Init/Data/BitVec/Lemmas.lean:718]: Setting up Lawful classes is not sufficient reason to change the implicitness. You can use, e.g.
```
instance : Std.IdempotentOp (α := BitVec n) (· ||| · ) where
  idempotent _ := BitVec.or_self
```
here with the original implicitness.

Comment (tobiasgrosser) [reply src/Init/Data/BitVec/Lemmas.lean]: OK, that makes sense.

**Comment by tobiasgrosser** [comment]
awaiting-review

**Pull request merged**